stopcovid/dialog/engine.py

```diff
@@ -257,12 +265,30 @@ def _advance_to_next_drill(
     def _update_schedule_requested(
         self, dialog_state: DialogState, base_args: Dict[str, Any]
     ) -> Optional[List[stopcovid.dialog.models.events.DialogEvent]]:
-        prompt = dialog_state.get_prompt()
-        if prompt is None:
-            if self.content_lower in ["schedule", "calendario", "horario"]:
-                return [SchedulingDrillRequested(**base_args)]
+        if self.content_lower in ["schedule", "calendario", "horario"]:
+            return [SchedulingDrillRequested(**base_args)]
         return None
 
+    def _name_change_drill_requested(self, dialog_state: DialogState, base_args: Dict[str, Any]):
+        if self.content_lower in ["name", "nombre"]:
+            return [NameChangeDrillRequested(**base_args)]
+        return None
+
+    def _language_change_drill_requested(
+        self, dialog_state: DialogState, base_args: Dict[str, Any]
+    ):
+        if self.content_lower in ["lang", "idioma"]:
+            return [LanguageChangeDrillRequested(**base_args)]
+        return None
+
+    def _menu_requested(self, dialog_state: DialogState, base_args: Dict[str, Any]):
+        if self.content_lower in ["menu", "menú"]:
+            return [MenuRequested(**base_args)]
+        return None
+
+    def _unknown_message_received(self, dialog_state: DialogState, base_args: Dict[str, Any]):
+        return [UnknownMessageReceived(**base_args, message=self.content)]
```

Comment:
I think "UnknownMessage" is a little too vague. Could refer to message content, or even a person. I'd be more onboard with something specific like "OutOfDrillMessageReceived" or "OutOfContextMessageReceived"

Reply:
I was considering "unhandled" instead of "unknown". What do you think of that?

Reply:
💯 thats perfect